# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/ServiceWorkerRegistration.webidl
# Commit: 9c8b6e35bf3f
# Full Hash: 9c8b6e35bf3f9916e114ced0473be4b778c7bab1
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r a9b3b52f33a5 -r 9c8b6e35bf3f dom/webidl/ServiceWorkerRegistration.webidl
--- a/dom/webidl/ServiceWorkerRegistration.webidl	Tue Sep 10 16:21:33 2024 +0000
+++ b/dom/webidl/ServiceWorkerRegistration.webidl	Tue Sep 10 16:31:38 2024 +0000
@@ -7,6 +7,7 @@
  * http://slightlyoff.github.io/ServiceWorker/spec/service_worker/index.html
  * https://w3c.github.io/push-api/
  * https://notifications.spec.whatwg.org/
+ * https://wicg.github.io/cookie-store/#idl-index
  */
 
 [Func="ServiceWorkerVisible",
@@ -52,3 +53,11 @@
   [NewObject, Func="mozilla::dom::Notification::PrefEnabled"]
   Promise<sequence<Notification>> getNotifications(optional GetNotificationOptions filter = {});
 };
+
+/* Bug 1475599 - We decide to do not implement the entire cookie-store spec.
+ * Instead, we implement only the subset that is compatible with document.cookie
+[Exposed=(ServiceWorker,Window)]
+partial interface ServiceWorkerRegistration {
+  [SameObject] readonly attribute CookieStoreManager cookies;
+};
+*/